api/controllers/user.go

```diff
@@ -92,3 +98,74 @@ func HandleSignUp(c *gin.Context) {
 
 	c.IndentedJSON(http.StatusOK, gin.H{"message": "User created successfully"})
 }
+
+func HandleLogIn(c *gin.Context) {
+	var err error
+	var form interfaces.LogInForm
+	var user interfaces.User
+
+	if err := c.BindJSON(&form); err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Bad request"})
+		return
+	}
+
+	_, err = mail.ParseAddress(form.Email)
+
+	//Check if exists empty fields
+	if form.Email == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Email cannot be empty"})
+		return
+	} else if err != nil { //Check email format
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Email is invalid"})
+		return
+	}
+
+	if form.Password == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Password cannot be empty"})
+		return
+	}
+
+	user, err = models.CheckExistEmail(form.Email)
+
+	//Check if exists email
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusUnauthorized,
+				gin.H{"message": "Wrong Email/Password"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError,
+				gin.H{"message": "Internal server error"})
+			return
+		}
+	}
+
+	//Check if user is verified
+	if !user.IsVerified {
+		c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"message": "User has not been verified"})
```

Comment:
I suggest to use the 403 (Forbidden) status code when the password is correct but the user is not verified.